### DISCUSSION TITLE
igir produced zip files do not preserve directories

### DISCUSSION THREAD

**Discussion by tnvu** [emmercm/igir#1001] [Q&A] [answered]
Here is an entry for a game that has subdirectories

```
        <game name="Castlevania - Harmony of Despair - TU 2 (World) (Addon)" id="06039">
                <description>Castlevania - Harmony of Despair - TU 2 (World) (Addon)</description>
                <game_id>6480C08F</game_id>
                <rom name="Cache\TU_1C422JQ_000010G000000.00000000000G4" size="2437120" crc="c90aa7c2" md5="8bb79b961fb0ea6371e6464ed0e73494" sha1="005b519c767c4d77998223296b4394882f49d259" sha256="983b6e84e4e2d209031b401f5dca4bb4578b18cb
904ecdc47691bc43eba03169"/>
        </game>
```

Here is the listing of the files in the zip created by igir

```
unzip -l output/Microsoft\ -\ Xbox\ 360\ \(Digital\)/Castlevania\ -\ Harmony\ of\ Despair\ -\ TU\ 2\ \(World\)\ \(Addon\).zip 
Archive:  output/Microsoft - Xbox 360 (Digital)/Castlevania - Harmony of Despair - TU 2 (World) (Addon).zip
  Length      Date    Time    Name
---------  ---------- -----   ----
  2437120  03-09-2024 02:03   Cache_TU_1C422JQ_000010G000000.00000000000G4
---------                     -------
  2437120                     1 file

```

Notice that the slashes are not preserved and converted to underscores.

I see in https://github.com/emmercm/igir/blob/main/src/types/files/archives/zip.ts#L208 that there is a conversion of backslashes to forward slashes, but I don't see where the forward slashes get converted to underscores?

Thanks

**Comment by tnvu** [comment]
I see it here but couldn't trace it back to something that made  sense to me.

https://github.com/emmercm/igir/blob/main/src/types/outputFactory.ts#L536

**Comment by emmercm** [comment]
I'm guessing that this is Linux still? It may be a Unix `/` vs. Windows `\` issue.

**Comment by tnvu** [comment]
Yeah, still linux... I wouldn't think it's a Unix v Windows issue..

It looks like all back and forward slashes are converted to forward slashes in 

https://github.com/emmercm/igir/blob/main/src/types/files/archives/zip.ts#L108

But then in the output factory, Dat.getRomNamesContainDirectories() returns false but it looks like it's true by default.

**Reply by emmercm** [reply, marked as answer]
This will fix it: https://github.com/emmercm/igir/pull/1016

**Comment by emmercm** [comment]
What DAT group is this from @tnvu? I can't seem to find it.

**Reply by emmercm** [reply]
Never mind, I found it, No-Intro `Microsoft - Xbox 360 (Digital)`: https://datomatic.no-intro.org/index.php?page=download&op=dat&s=57
